index.js

```diff
@@ -0,0 +1,174 @@
+const commandLineArgs = require('command-line-args');
+const fs = require('fs');
+const mongoose = require('mongoose');
+const readline = require('readline');
+
+
+
+
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+mongoose.Promise = global.Promise;
+mongoose.connect('mongodb://localhost:21078/knapsack');
+
+const KnapSchema = new mongoose.Schema({
+    _id: { type: Number, required: true },
+    weight: { type: Number, required: true },
+    value: { type: Number, required: true }
+});
+const Knap = mongoose.model('Knap', KnapSchema);
+function createKnap(o) {
+    const { id, weight, value } = o;
+    const newKnap = new Knap(_id = id, weight, value);
+    newKnap.save().then(
+
```

Comment:
If nothing goes here you can remove the then part

Reply:
Thought I might need MongoDB to do the work, instead it helped me debug because I queried to see the transitions to the final state for the 2 weight column.